### DISCUSSION TITLE
Batch Insert on factory()->count()->create()

### DISCUSSION THREAD

**Discussion by JHWelch** [laravel/framework#44975] [Ideas]
As was pointed out by [Ryuta Hamasaki on twitter](https://twitter.com/avosalmon/status/1592792095022280704) currently when you call 

```php
User::factory()->count(100)->create();
```

It will execute 5 separate insert statements. So if you want to execute a single insert with 100 records you have to use something like

```php
User::insert(
    User::factory()->count(100)->make()->toArray();
);
```

Which is obviously much more verbose. Is there are a reasons factories are constructed like this? I was curious if I could PR this, but didn't want to go too far down the rabbit hole if there was a good reason for this behavior. I would assume it may have something to do with the other more advanced features of factories that may not work as well trying to batch insert them.

**Comment by tyabu12** [comment]
I think that the reason is that the model creating/created/saving/saved events do not fire on `User::insert`.

**Reply by tyabu12** [reply]
And model relations on save also be relevant.

**Reply by JHWelch** [reply]
Ahh that makes sense to me. It needs to use `save` to trigger the actions properly. Think you could also batch the relations if you did enough fiddling, but that would be pretty hard.
